benchmark/misc/util-extend-vs-object-assign.js

```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+const common = require('../common.js');
+const util = require('util');
+const v8 = require('v8');
+
+const bench = common.createBenchmark(main, {
+  type: ['extend', 'assign'],
+  n: [10e4]
+});
+
+function main(conf) {
+  let fn;
+  const n = conf.n | 0;
+  let v8command;
+
+  if (conf.type === 'extend') {
+    fn = util._extend;
+    v8command = '%OptimizeFunctionOnNextCall(util._extend)';
+  } else if (conf.type === 'assign') {
```

Comment:
FYI: https://github.com/nodejs/node/blob/master/benchmark/common.js#L255

the benchmark/common.js now has a method for handling the details of v8 optimization for you. e.g.

```js
function myMethod(a,b) {
  /** ... **/
}
common.v8ForceOptimization(myMethod, 'a', 'b');
myMethod('a', 'b');
```